### DISCUSSION TITLE
AULA 3.10-PRIMEIRO SCRIPT

### DISCUSSION THREAD

**Discussion by hebermerce** [chgasparoto/curso-aws-com-terraform#117] [Geral]
Boa noite.

estou tentando configurar o bucket porém ele retorna com essa mensagem.
No changes. Your infrastructure matches the configuration.

Terraform has compared your real infrastructure against your configuration and found no differences, so no changes are needed.

segue print abaixo

**Comment by chgasparoto** [comment]
@hebermerce eu removi a imagem pois vc estava mostrando suas credenciais. Sugiro vc deletar o usuário que está usando e criar um novo para gerar novas credencias de acesso.

Sobre a mensagem de erro, tente alterar ou adicionar uma tag ou o nome do bucket. Está falando que não tem nenhuma alteração para aplicar na AWS.

**Reply by MarceloPaglione2411** [reply]
tive o mesmo problema do nosso amigo acima. Alterei a tag e mesmo assim nao funciona.

**Reply by MarceloPaglione2411** [reply]
estranho que ele nao criou o arquivo .hcl quando executei o terraform init

**Reply by chgasparoto** [reply]
@988094843 consegue postar uma screenshot da estrutura da sua pasta e da mensagem após rodar o comando?

**Reply by MarceloPaglione2411** [reply]
na pasta nao cria a estrutura de arquivos, nota que so tem o .tf

**Comment by MarceloPaglione2411** [comment]
![image](https://user-images.githubusercontent.com/104773898/188597859-b045e244-8320-4b58-bc40-010eff592840.png)

![image](https://user-images.githubusercontent.com/104773898/188598128-edaa8435-5791-45e4-a403-90db4d9702ee.png)

**Reply by chgasparoto** [reply]
Qual a versão do seu terraform? O arquivo de lock só é gerado a partir da versão `0.14`.

E tenta acionar mais uma tag no block de `tags`.

Exemplo:
```hcl
tags = {
  ...
 CreatedAt = "2022-09-06"
}
```

**Reply by MarceloPaglione2411** [reply]
vou lhe passar todos os dados. verifica se tem algo de errado, pois continua com o mesmo problema. Detalhe... coloquei a tag que vc pediu.
![image](https://user-images.githubusercontent.com/104773898/188880440-3e0b55bd-fc22-441f-a301-b3073e934f6f.png)
![image](https://user-images.githubusercontent.com/104773898/188880640-0ddbaf6d-ef2a-47f1-b186-6f2d288e93a8.png)

**Reply by MarceloPaglione2411** [reply]
segue a versao
![image](https://user-images.githubusercontent.com/104773898/188882762-0d20ef90-1a27-4048-87d9-04d2ffa6cb42.png)

**Reply by chgasparoto** [reply]
Ok, no windows os arquivos que começam com `.` não ficam ocultos por padrão? Talvez seja isso, o arquivo está lá mas você não consegue ver pois ele está oculto.

**Reply by MarceloPaglione2411** [reply]
entao nao deu certo..... infelizmente isso esta impedindo de fazer as aulas na pratica.  teria outro jeito de vc me ajudar ou auxiliar??

**Reply by chgasparoto** [reply]
posta um video ou um gif de vc executando os comandos, assim eu consigo ver certinho o seu workflow

**Reply by MarceloPaglione2411** [reply]
segue o video

https://user-images.githubusercontent.com/104773898/189239109-09c06710-8173-4726-b7d6-cf137353925f.mp4

![WhatsApp Image 2022-09-08 at 19 45 42](https://user-images.githubusercontent.com/104773898/189239115-39ff199a-0846-4d17-83ca-7bdf44743e82.jpeg)

**Reply by chgasparoto** [reply]
Não consigo ver nada de errado com a sequência dos comandos. É esperado que retorna esta mensagem se você não fizer nenhuma alteração no arquivo comparado com o que você já tem na AWS.

Uma coisa é que o seu arquivo não está salvo com as últimas alterações e o bloco de terraform está com erro de digitação.

Tenta fazer exatamente o que você fez mas troque o `cidr_block` para `10.1.1.0/16` dentro do bloco da `vpc`. Salve o arquivo e rode `terraform plan`. É para você ver o diff no terminal.

**Reply by MarceloPaglione2411** [reply]
ok!

**Reply by MarceloPaglione2411** [reply]
adicionei esse codigfo no cabeçalho main.tf e deu certo!!!  

```hcl
terraform {
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = ">= 4.29.0"
    }
  }
}
```
